src/models/methods/subscribe.ts

```diff
@@ -33,16 +33,16 @@ interface BaseStream {
   type: string
 }
 
-export interface LedgerStream extends BaseStream {
-  type: 'ledgerClosed'
+export interface LedgerStream {
+  type?: 'ledgerClosed'
```

Comment:
Why is this type optional?

Reply:
The type is optional here because the response from a 'ledger' `SubscribeRequest` omits the type, but later updates on the `LedgerStream` include the type.

Here's the docs and quote: https://xrpl.org/subscribe.html#response-format
> Stream: ledger - Information about the ledgers on hand and current fee schedule. This includes the same fields as a ledger stream message, except that it omits the type and txn_count fields.


Would a better solution be to have a separate type that's just for the response from a ledger subscribe request?

Reply:
That response is actually different from the actual stream. That is what the `subscribe` request returns, which is different than what the listener is listening to.